docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+# Immutability in JavaScript
+
+According to (wikipedia)[https://en.wikipedia.org/wiki/Immutable_object] an *immutable object (unchangeable object) is an object whose state cannot be modified after it is created*. This rule is quite simple, if you want to modify some property of an object you have to do it on a copy. We will see a bit later what improvements and fancy features it unlocks for our developments.
+
+## EcmaScript
+
+EcmaScript provides utilities to keep our data immutable. Arrays and objects APIs contain methods to create copies and even prevent instances from being updated. More recently EcmaScript introduced a new syntaxes to create copies of objects and arrays.
+
+### Object.assign
+
+We want to add a name property in our object.
+
+https://gist.github.com/frinyvonnick/a281f15288d3ab90ce5354f01df52ccf
+
+We can do it with `Object.assign` and a little trick. Basically it copies all properties from an object into another one, thus it mutates the target object. We use a small trick, passing an empty object as first parameter, which creates a new reference.
+
+https://gist.github.com/frinyvonnick/98878e45d6a8ccb4618238dbec175ae9
+
+We now have a new object with our new name property and a commonNames property remain unchanged. With this method you can create/overwrite multiple properties at the same time.
```

Comment:
Tu ne montres pas le résultat ? 🤔 
je mettrai peut-être 
name property => `name` property
a commonNames property => the `commonNames` property